File: client/web/sudoo/lib/domain/repository/storage_repository.dart
import 'package:sudoo/domain/core/data_state.dart';

abstract class StorageRepository {
  Future<DataState<String, Exception>> uploadImage(String pathImage);
  Future<DataState<String, Exception>> uploadImageBytes(List<int> bytes, {String? imageName});
  Future<DataState<String, Exception>> uploadFileBytes(List<int> bytes, {String? fileName});
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/repository/StorageRepository.kt
package com.sudo248.sudoo.domain.repository

import android.content.Context
import android.net.Uri
import com.sudo248.base_android.core.DataState
import java.io.File

interface StorageRepository {
    suspend fun uploadImage(context: Context, uri: Uri): DataState<String, Exception>
    suspend fun uploadImage(pathImage: String): DataState<String, Exception>

    suspend fun getArModelResource(parent: File, source: String): DataState<Uri, Exception>
}